develop/app/Controllers/v2/PaymentController.php

```diff
@@ -0,0 +1,262 @@
+            "paymentID" => $paymentCreatedIDOrNull,
+            "msg"       => "Payment create method successful."
+        ]);
+    }
+
+    /**
+     * [PUT] /api/v1/payment
+     * Update payment price.
+     *
+     * @param int $paymentKey
+     * @return void
+     */
+    public function update($paymentKey = null)
+    {
+        $data = $this->request->getJSON(true);
+
+        if (is_null($paymentKey)) {
+            return $this->fail("The payment key is required.", 404);
+        }
+
+        if (is_null($data["total"]) || is_null($paymentKey)) {
+            return $this->fail("Incoming data not true", 400);
+        }
+
+        $total  = $data["total"];
+        $status = $data["status"] ?? "paymentUpdate";
+
+        $paymentModel  = new PaymentModel();
+
+        $paymentEntity = $this->getPaymentEntityByPaymentKey($paymentKey);
+        if (is_null($paymentEntity)) {
+            return $this->fail("This payment information is not exist or cannot found.", 404);
+        }
+
+        $result = $paymentModel->where('pm_key', $paymentKey)
+                               ->set('total', $total)
+                               ->set('status', $status)
+                               ->update();
+
+        if ($result) {
+            return $this->respond([
+                "status" => true,
+                "msg"    => "Payment update method successful."
+            ]);
+        } else {
+            return $this->fail("Payment update method fail", 400);
+        }
+    }
+
+    /**
+     * [DELETE] /api/v1/payment/{paymentKey}
+     * Delete payment.
+     *
+     * @param int $paymentKey
+     * @return void
+     */
+    public function delete($paymentKey = null)
+    {
+        if (is_null($paymentKey)) {
+            return $this->fail("The payment key is required.", 404);
+        }
+
+        $paymentModel  = new PaymentModel();
+
+        $paymentEntity = $this->getPaymentEntityByPaymentKey($paymentKey);
+        if (is_null($paymentEntity)) {
+            return $this->fail("This payment information is not exist or cannot found.", 404);
+        }
+
+        $setDeleteStatus = $paymentModel->where('pm_key', $paymentKey)
+                                        ->set("status", "PaymentDelete")
+                                        ->update();
+        if (!$setDeleteStatus) {
+            return $this->fail("This payment status change to 'DELETE' fail.", 400);
+        }
+
+        $result = $paymentModel->delete($paymentKey);
+
+        if ($result) {
+            return $this->respond([
+                "status" => true,
+                "msg"    => "Payment delete method successful."
+            ]);
+        } else {
+            return $this->fail("Payment delete fail", 400);
+        }
+    }
+
+    /**
+     * Get payment entity by payment key
+     *
+     * @param int $payment_key
+     * @return PaymentEntity|null
+     */
+    public function getPaymentEntityByPaymentKey($payment_key = null): ?PaymentEntity
```

Comment:
Re-name this method and limit the type of incoming data like: `int $payment_key`.

Reply:
This change is done.